code/game/machinery/iv_drip.dm

```diff
@@ -1,224 +1,700 @@
+	return ..()
 
+/obj/machinery/iv_drip/update_icon()
+	cut_overlays()
 	if(beaker)
+		add_overlay("beaker[tipped ? "_tipped" : ""]")
 		var/datum/reagents/reagents = beaker.reagents
 		if(reagents?.total_volume)
-			var/image/filling = image('icons/obj/iv_drip.dmi', src, "reagent")
-
+			var/image/filling = image('icons/obj/iv_drip.dmi', src, "[tipped ? "tipped_" : ""]reagent")
 			var/percent = round((reagents.total_volume / beaker.volume) * 100)
 			switch(percent)
-				if(0 to 9)		filling.icon_state = "reagent0"
-				if(10 to 24) 	filling.icon_state = "reagent10"
-				if(25 to 49)	filling.icon_state = "reagent25"
-				if(50 to 74)	filling.icon_state = "reagent50"
-				if(75 to 79)	filling.icon_state = "reagent75"
-				if(80 to 90)	filling.icon_state = "reagent80"
-				if(91 to INFINITY)	filling.icon_state = "reagent100"
+				if(0 to 9)		filling.icon_state = "[tipped ? "tipped_" : ""]reagent0"
+				if(10 to 24) 	filling.icon_state = "[tipped ? "tipped_" : ""]reagent10"
+				if(25 to 49)	filling.icon_state = "[tipped ? "tipped_" : ""]reagent25"
+				if(50 to 74)	filling.icon_state = "[tipped ? "tipped_" : ""]reagent50"
+				if(75 to 79)	filling.icon_state = "[tipped ? "tipped_" : ""]reagent75"
+				if(80 to 90)	filling.icon_state = "[tipped ? "tipped_" : ""]reagent80"
+				if(91 to INFINITY)	filling.icon_state = "[tipped ? "tipped_" : ""]reagent100"
 
 			var/reagent_color = reagents.get_color()
 			filling.icon += reagent_color
 			add_overlay(filling)
+		if(attached)
+			add_overlay("iv_in[tipped ? "_tipped" : ""]")
+			if(mode)
+				add_overlay("light_green[tipped ? "_tipped" : ""]")
+			else
+				add_overlay("light_red[tipped ? "_tipped" : ""]")
+		else
+			add_overlay("iv_out[tipped ? "_tipped" : ""]")
+	if(tank)
+		if(istype(tank, /obj/item/tank/oxygen))
+			tank_type = "oxy"
+		else if(istype(tank, /obj/item/tank/anesthetic))
+			tank_type = "anest"
+		else if(istype(tank, /obj/item/tank/phoron))
+			tank_type = "phoron"
+		else
+			tank_type = "other"
+		add_overlay("tank_[tank_type][tipped ? "_tipped" : ""]")
+		update_gauge()
+	if(breath_mask)
+		if(breather)
+			add_overlay("mask_on[tipped ? "_tipped" : ""]")
+		else
+			add_overlay("mask_off[tipped ? "_tipped" : ""]")
+		if(epp_active)
+			add_overlay("light_blue[tipped ? "_tipped" : ""]")
+
+/obj/machinery/iv_drip/proc/update_gauge()
+	var/gauge_pressure = 0
+	var/last_gauge_pressure
+	if(tank.air_contents)
+		gauge_pressure = tank.air_contents.return_pressure()
+		if(gauge_pressure > 1015) // tanks.dm TANK_IDEAL_PRESSURE 1015
+			gauge_pressure = -1
+		else
+			gauge_pressure = round((gauge_pressure/1015)*tank.gauge_cap)
+	if(gauge_pressure == last_gauge_pressure)
+		return
+	last_gauge_pressure = gauge_pressure
+	add_overlay("[tank.gauge_icon][(gauge_pressure == -1) ? "overload" : gauge_pressure][tipped ? "_tipped" : ""]")
 
-/obj/machinery/iv_drip/MouseDrop(over_object, src_location, over_location)
-	..()
+/obj/machinery/iv_drip/machinery_process()
+	set background = 1
```

Comment:
```suggestion

```
Don't do this, the machinery subsystem will handle it.

Reply:
Surprisingly enough that was in the original IV code. I didn't know whether to remove it or not, so I decided to not touch it as I was re-organizing the code by the end. I'll remove it shortly